ecosystem/sep-0024.md

```diff
@@ -170,26 +171,128 @@ There are several possible kinds of response, depending on whether the anchor ne
+   * The single endpoint needs the `claimable_balance_id` in order to retrieve the claimable balance info.
+     * This can be obtained by extracting the `claimable_balance_id` from `results_xdr` of the deposit transaction submitted if the anchor does not extend their [transaction history endpoints.](#anchor-feature-extend-transaction-history-endpoints-for-claimable-balances)
+
+##### Wallet Feature: Claim Account Claimable Balance
+
+It's recommended that Wallets provides a UI to enable accounts to claim their balances [polled](#wallet-feature-claimable-balance-polling). Accounts will still need to establish trustline to the asset before claiming the outstanding balance.
+
+**NOTE:** The account will only need to establish a trustline to the asset they wish to claim once before claiming.
+
+Example on how to claim a balance. Omitted from the example is setting trust operation.
+
+```javascript
+  const transaction = new TransactionBuilder(account, {
+      fee: 100,
+      networkPassphrase: this.network_passphrase,
+    })
+      .addOperation(
+        Operation.claimClaimableBalance({
+          balanceId,
+        })
+      )
+      .setTimeout(0)
+      .build()
+  transaction.sign(keypair)
+  const result = await this.server.submitTransaction(transaction)
+```
+
+##### Anchor Claimable Balance Flow
+
+1. Wallets make a request to `/transactions/deposit/interactive` providing the `claimable_balance_supported` request parameter.
+2. Anchors update their internal database record of the transaction to indicate the wallet supports receiving a [claimable balance.](#deposit)
+3. User sends external assets to the anchor's off-chain account.
+4. Anchors detect that the user's Stellar account doesn't have a trustline.
+5. From there the anchor will continue the [deposit flow](#basic-anchor-implementation), crafting the transaction as a claimable balance operation.
+
+##### Anchor Feature: Extend Transaction history endpoints For Claimable Balances
+
+Wallets query the [transaction endpoint](#transaction-history) to see and display the status of deposits.
+Anchors should extend their transaction(s) endpoint response to include the `claimable_balance_id` that the anchor added to the [transaction field.](####Fields-for-deposit-transactions)
+This will help wallets detect that the user's deposit was successful and theres an [available claimable balance for them.](#wallet-feature-claimable-balance-polling)
+
+##### Anchor Claimable Balance Claimants and Predicates
+
+Predicates are one of the [claimable balance parameters](https://developers.stellar.org/docs/glossary/claimable-balance/#parameters) used to craft a Claimable Balance transaction. They're conditions to meet in order for the recipient to claim their balance.
+
+It's recommended to use the `UNCONDITIONAL` predicate to allow the claimant to claim the balance without any barriers.
```

Comment:
Why is putting some sane predicates (collect within 30 days, for example) not recommended?

Reply:
If the account doesn't claim the balance in 30 days the balance unfortunately doesn't refund back into the sender's account. It's lost for both parties.
In order to provide a friendlier experience for the claimant/receiver its best not to restrict them from when they want to establish their trustline and claim.

If the claimant cannot claim the balance due to time constraints or whatnot then we would have to recommend the following for Wallets and Anchors

- Wallets need to provide more aggressive messaging to their users that there's a deadline to claim their balance
- Wallets and Anchors need to provide ways to refund deposits in the scenario where a claimablebalance is dead on the DEX
- (or) Anchors need refund themselves by issuing more of their asset on the network to send another claimable balance to the accounts that forget to claim in 30 days

Reply:
I think the simplest option would to have the sponsor list themselves as a claimant, and claim the balance if it's outstanding for too long.

Reply:
I think thats an interesting idea.
Maybe we should see what the community thinks over time.

Reply:
I prefer not having a recommendation rather than recommending something that we're not sure is the best practice

Reply:
we can leave it up for the anchor for now

Reply:
@tomerweller  what do you think of this?
```suggestion
Anchors are free to set whichever predicates they feel are necessary in order to claim the balance.
If there are no predicate preferences, `UNCONDITIONAL` allows accounts to claim balances at anytime.
```

Reply:
LGTM!